### PR TITLE
DCOS-19864: Fix select

### PR DESCRIPTION
**Pull request by juliangieseke**
This PR fixes the issue, that the select wasnt rerender itself when props changed.

### CODE REVIEW

**Comment by juliangieseke** [comment]
well…
screen shot 2017-12-13 at 13 00 47

**Comment by juliangieseke** [comment]
Btw: we can get rid of `<SelectOption> if we somehow force all children to have a `value` prop… 🤔

**Comment by ahoskins** [comment]
@juliangieseke I don't know much about why `SelectOption` is there, but I'd say it's out of this PR scope, wouldn't you agree? I think this PR should be the minimal set of work to make Select work, and avoid changing it's API.

File: src/Select/Select.js [hunk 1]
```diff
@@ -64,12 +35,16 @@ export default class Select extends React.Component {
...
             display: "none"
           }}
-          value={this.state.value}
+          value={this.props.value}
           onChange={this.handleInputChange.bind(this)}
         />
         <Dropdown
           items={this.buildItemsArray()}
-          initialID={this.state.value}
+          persistentID={
```
Comment (ahoskins) [inline src/Select/Select.js]: Nice, didn't notice this as a possible situation 👍

File: src/Select/__tests__/Select-test.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ describe("Select", function() {
         className="select-class-1 select-class-2"
         onChange={this.callback}
         name="SelectName"
+        value="option-3"
```
Comment (ahoskins) [inline src/Select/__tests__/Select-test.js]: 😄

**Comment by juliangieseke** [comment]
@ahoskins
re `SelectOption`: you're right, i have changed as much as I could argue is prt of the fix, but removing the `SelectOption` would be out of scope.

---

+1 to andrews last comment

**Pull request merged**